### PR TITLE
[NVIDIA] Add new activation, comparison and reduce operations

### PR DESCRIPTION
**Pull request by nkogteva**
### Details:
- *Support the following activations: Abs, Cos, Cosh, Elu, Exp, Gelu, HSwish, Sin, Sinh, Sqrt, Log, Mish*
- *Support the following comparison operations: GreaterEqual, LessEqual, Equal, NotEqual*
- *Support the following reduce operations: ReduceMax, ReduceMean, ReduceMin, ReduceProd*
- *Update StridedSlice to support i32 input*

Depends on https://github.com/openvinotoolkit/openvino/pull/19532

### Tickets:
- [106571](https://jira.devtools.intel.com/browse/CVS-106571)
- [106572](https://jira.devtools.intel.com/browse/CVS-106572)
- [106573](https://jira.devtools.intel.com/browse/CVS-106573)
- [106576](https://jira.devtools.intel.com/browse/CVS-106576)
- [106579](https://jira.devtools.intel.com/browse/CVS-106579)
- [106585](https://jira.devtools.intel.com/browse/CVS-106585)

### CODE REVIEW

File: modules/nvidia_plugin/src/kernels/elu.cu [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "elu.hpp"
+
+namespace ov {
+namespace nvidia_gpu {
+namespace kernel {
+
+namespace cumath = CUDA::math;
+template <typename T>
+struct EluOpImpl {
+    __device__ static inline T op(T x, double alpha) {
```
Comment (vladimir-paramuzov) [inline modules/nvidia_plugin/src/kernels/elu.cu:17]: Why double? float would be enough

File: modules/nvidia_plugin/src/transformer/reduce_transformation.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    const ov::Shape new_output_shape = new_reduce->output(0).get_shape();
+    if (input_shape != new_output_shape) {
+        auto reshape_const = std::make_shared<ov::op::v0::Constant>(element::i32, Shape{output_shape.size()}, output_shape);
+        auto reshape = std::make_shared<ov::op::v1::Reshape>(new_reduce, reshape_const, false);
+        for (auto consumer : consumers) {
+            consumer.replace_source_output(reshape);
+        }
+        new_ops = {new_reduce, reshape_const, reshape};
+    } else {
+        auto reshape_const = std::make_shared<ov::op::v0::Constant>(element::i32, Shape{output_shape.size()}, output_shape);
```
Comment (vladimir-paramuzov) [inline modules/nvidia_plugin/src/transformer/reduce_transformation.cpp:72]: That can be probably replaced with ConvertReduceToReshape common transformation call

File: modules/nvidia_plugin/src/transformer/reduce_transformation.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        auto reshape = std::make_shared<ov::op::v1::Reshape>(reduce->input_value(0), reshape_const, false);
+        for (auto consumer : consumers) {
+            consumer.replace_source_output(reshape);
+        }
+        new_ops = {reshape_const, reshape};
+    }
+    ov::copy_runtime_info(reduce, new_ops);
+    for (auto& new_op : new_ops) {
+        new_op->get_rt_info()[ExecGraphInfoSerialization::ORIGINAL_NAMES] = reduce->get_friendly_name();
```
Comment (vladimir-paramuzov) [inline modules/nvidia_plugin/src/transformer/reduce_transformation.cpp:81]: I think this code is not needed

Comment (nkogteva) [reply modules/nvidia_plugin/src/transformer/reduce_transformation.cpp:81]: Separate PR with cleanup https://github.com/openvinotoolkit/openvino_contrib/pull/715

Comment (nkogteva) [reply modules/nvidia_plugin/src/transformer/reduce_transformation.cpp]: Removed

File: modules/nvidia_plugin/src/transformer/reduce_transformation.cpp (hunk 2 above)
Comment (nkogteva) [reply modules/nvidia_plugin/src/transformer/reduce_transformation.cpp]: PR with support `ReduceProd` in `ConvertReduceToReshape` https://github.com/openvinotoolkit/openvino/pull/19532

File: modules/nvidia_plugin/src/kernels/elu.cu (hunk 1 above)
Comment (nkogteva) [reply modules/nvidia_plugin/src/kernels/elu.cu]: Replaced by float

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**